static/main.js

```diff
@@ -10,25 +10,36 @@ function init() {
     }
 
     if ("addEventListener" in permalink) {
-        permalink.addEventListener("click", function() {
-            window.location.search = "?text=" + encodeURIComponent(input.value);
-        });
+        permalink.addEventListener("click", setSearch);
     } else {
-        permalink.attachEvent("click", function() {
-            window.location.search = "?text=" + encodeURIComponent(input.value);
-        });
+        permalink.attachEvent("click", setSearch);
     }
 
     var match = (/(?:^\?|&)text=([^&]*)/).exec(window.location.search);
     if (match) {
         input.value = decodeURIComponent(match[1]);
     }
 
+    var macros = {};
+    var options = {};
+    var macroRegex = /(?:^\?|&)(?:\\|%5[Cc])([A-Za-z]+)=([^&]*)/g;
+    var macroString = "";
+    while ((match = macroRegex.exec(window.location.search)) !== null) {
+        options.macros = macros;
+        macros["\\" + match[1]] = decodeURIComponent(match[2]);
+        macroString += "&" + match[0].substr(1);
+    }
```

Comment:
nice addition, I checked out your branch locally to try this out and it works as advertised